bokeh/models/renderers.py

```diff
@@ -45,6 +48,28 @@ def _check_bad_column_name(self):
         if missing:
             return "%s [renderer: %s]" % (", ".join(sorted(missing)), self)
 
+    @validation.warning(COLON_IN_CATEGORY_LABEL)
+    def _check_colon_in_category_label(self):
+        if not self.glyph: return
+        if not self.data_source: return
+        vm = self.glyph.vm_serialize()
+        labels = (label for label in ['x', 'y']
+                  if label in vm and 'field' in vm[label])
```

Comment:
Poked around a bit, there is no good programmatic way currently to distinguish the coordinate data specs from other ones. But the set of names that are used as coordinates are unique I believe, so maybe turn the test around? 
```
labels = []
for key, val in vm.items():
    if val.get('field', None) in  ['x', 'y','xs','ys', ...... ] # all the rest in this list
        labels.append(val['field'])
```
Not great but hopefully this test can be removed by 0.11 when the restriction is removed.